### ISSUE TITLE
Pass pre-downloaded repo to dependency fetcher to avoid downloading dependencies on every rebuild

### ISSUE THREAD

**Issue by shadowrylander** [DavHau/mach-nix#177]
Hello!

Essentially what the title says; how could I pass the output of, say, a fetchGit or flake input to mach-nix to be used in place of a newly-downloaded dependency repo? This will prevent a redownload every time I rebuild my system.

Thank you kindly for the help!

**Comment by DavHau** [comment]
One specific version of mach-nix always uses the same version of dependency data, unless you specify the dependency data revision manually via `pypiDataRev` and `pypiDataSha256`.

In any case the dependency data is cached between by nix.
It will only be gone, if either:
  1. You upgrade mach-nix to a new ersion (-> will use newer deps data)
  2. You run a nix garbage collection.

Not sure which of these two you're running into.

For problem No. 2:
I could add an option like `preventGarbageCollect` to mach-nix, which will link the deps data into your derivation, so it wouldn't get garbage collected. At least not until you remove that mach-nix environment from your configuration.

Would that help? Or did I misunderstand your problem?

**Comment by shadowrylander** [comment]
I don't quite know for the first one; I've set up automatic garbage collection, but it's supposed to run weekly. And since I'm using flakes, the mach-nix version isn't changing either.

I think implementing `preventGarbageCollect` would help; I'm not quite sure how to debug this otherwise.

**Comment by DavHau** [comment]
Closing in favor of: https://github.com/DavHau/mach-nix/issues/181

**Comment by shadowrylander** [comment]
Thanks! I'll be following that, then!
